code/game/objects/items/weapons/implants/vorestuff_vr.dm

```diff
@@ -0,0 +1,131 @@
+	var/random_emote = null //An emote the person with the implant may be forced to perform after a prob check, such as [X] meows.
+	var/assigned_proc = /mob/living/carbon/human/proc/use_reagent_implant
+
+/obj/item/weapon/implant/reagent_generator/tempest
+	generated_reagent = "milk"
+	usable_volume = 1000
+
+	empty_message = "Your breasts are almost completely drained!"
+	full_message = "Your teats feel heavy and swollen!"
+	emote_descriptor = "squeezes milk"
+	self_emote_descriptor = "squeeze"
+	random_emote = "moos"
+	assigned_proc = /mob/living/carbon/human/proc/use_reagent_implant_tempest
+
+/obj/item/weapon/implant/reagent_generator/New()
+	..()
+	internal_storage = new /obj/item/weapon/reagent_containers/glass/beaker/large (src)
+	internal_storage.volume = usable_volume
+	internal_storage.reagents.maximum_volume = usable_volume
+
+/obj/item/weapon/implanter/reagent_generator/New()
+	..()
+	imp = new /obj/item/weapon/implant/reagent_generator(src)
+	update()
+	return
+
+/obj/item/weapon/implanter/reagent_generator/tempest/New()
+	..()
+	imp = new /obj/item/weapon/implant/reagent_generator/tempest(src)
+	update()
+	return
+
+/obj/item/weapon/implant/reagent_generator/implanted(mob/living/carbon/source)
+	processing_objects += src
+	to_chat(source, "<span class='notice'>You implant [source] with \the [src].</span>")
+	source.verbs |= assigned_proc
+	return 1
+
+/obj/item/weapon/implant/reagent_generator/process()
+	var/before_gen
+	if(imp_in && generated_reagent)
+		before_gen = internal_storage.reagents.total_volume
+		if(internal_storage && internal_storage.reagents.total_volume < internal_storage.volume)
+			if(imp_in.nutrition >= gen_cost)
+				do_generation()
+	else
+		imp_in.verbs -= assigned_proc
+		return
+
+	if(internal_storage)
+		if(internal_storage.reagents.total_volume == internal_storage.volume * 0.05)
+			to_chat(imp_in, "<span class='notice'>[empty_message]</span>")
+		else if(internal_storage.reagents.total_volume == internal_storage.volume && before_gen < internal_storage.volume)
+			to_chat(imp_in, "<span class='warning'>[full_message]</span>")
+
+/obj/item/weapon/implant/reagent_generator/proc/do_generation()
+	imp_in.nutrition -= gen_cost
+	internal_storage.reagents.add_reagent(generated_reagent, gen_amount)
+
+/mob/living/carbon/human/proc/use_reagent_implant()
+	set name = "Transfer From Reagent Implant"
+	set desc = "Remove reagents from am internal reagent into a container."
+	set category = "Object"
+	set src in view(1)
+
+	do_reagent_implant()
+
+/mob/living/carbon/human/proc/use_reagent_implant_tempest()
+	set name = "Milk"
+	set desc = "Grab Tempest's nipples and milk them into a container! May cause blushing and groaning."
+	set category = "Object"
+	set src in view(1)
+
+	do_reagent_implant()
+
+/mob/living/carbon/human/proc/do_reagent_implant()
+	if(!isliving(usr) || !usr.canClick())
+		return
+
+	if(usr.stat == 1 || (usr.restrained()))
+		return
```

Comment:
This stuff can use incapacitated() and pass it one of the several incap flags instead.